src/ServerRequestFactory.php

```diff
@@ -132,29 +153,47 @@ private function getUri(array $server): UriInterface
         return $uri;
     }
 
-    /**
-     * @suppress PhanUndeclaredFunction
-     */
     private function getHeadersFromGlobals(): array
     {
-        if (\function_exists('\getallheaders')) {
-            $headers = \getallheaders();
-            if ($headers === false) {
-                $headers = [];
+        if (function_exists('getallheaders')) {
+            $headers = getallheaders();
+            return $headers === false ? [] : array_filter($headers, static fn (string $value): bool => $value !== '');
+        }
+
+        $headers = [];
+
+        foreach ($_SERVER as $name => $value) {
+            if ($value === '') {
+                continue;
             }
-        } else {
-            $headers = [];
-            foreach ($_SERVER as $name => $value) {
-                if (strncmp($name, 'HTTP_', 5) === 0) {
-                    $name = str_replace(' ', '-', strtolower(str_replace('_', ' ', substr($name, 5))));
-                    $headers[$name] = $value;
+
+            if (strncmp($name, 'REDIRECT_', 9) === 0) {
+                $name = substr($name, 9);
+
+                if (array_key_exists($name, $_SERVER)) {
+                    continue;
                 }
             }
+
+            if (strncmp($name, 'HTTP_', 5) === 0) {
+                $headers[$this->normalizeHeaderName(substr($name, 5))] = $value;
+                continue;
+            }
+
+            if (strncmp($name, 'CONTENT_', 8) === 0) {
+                $headers[$this->normalizeHeaderName($name)] = $value;
```

Comment:
These values can be empty

Reply:
Yes

Reply:
Why add empty values when even the header for them was not sent by the client?

Reply:
Empty values will not be added. Look carefully at the code!

Reply:
https://github.com/yiisoft/yii-web/pull/357/files#r584582911